finesse/accuracy_functions.py

```diff
@@ -4,52 +4,127 @@
 import os
 from collections import namedtuple
 import regex as re
+from finesse.bing_search import BingSearch
+from dotenv import load_dotenv
 
 OUTPUT_FOLDER = "./finesse/output"
 AccuracyResult = namedtuple("AccuracyResult", ["position", "total_pages", "score"])
 
-def calculate_accuracy(responses_url: list[str], expected_url: str) -> AccuracyResult:
+def calculate_accuracy(responses_url: list[str], expected_url: list | str) -> AccuracyResult:
+    """
+    Calculates the accuracy of the responses by comparing the URLs of the responses with the expected URL.
+
+    Args:
+        responses_url (list[str]): A list of URLs representing the responses.
+        expected_url (list[str] | str): The expected URL or a list of expected URLs.
+
+    Returns:
+        AccuracyResult: An object containing the position, total pages, and score of the accuracy calculation.
+    """
     position: int = 0
     total_pages: int = len(responses_url)
     score: float = 0.0
-    expected_number = int(re.findall(r'/(\d+)/', expected_url)[0])
+    expected_number = []
+
+    PATTERN = r'/(\d+)/'
+    if isinstance(expected_url, list):
+        for url in expected_url:
+            if url.startswith("https://inspection.canada.ca"):
+                number = int(re.findall(PATTERN, url)[0])
+                expected_number.append(number)
+    elif isinstance(expected_url, str) and expected_url.startswith("https://inspection.canada.ca"):
+        number = int(re.findall(PATTERN, expected_url)[0])
+        expected_number.append(number)
 
     for idx, response_url in enumerate(responses_url):
-        response_number = int(re.findall(r'/(\d+)/', response_url)[0])
-        if response_number == expected_number:
-            position = idx
-            score = 1 - (position / total_pages)
-            score = round(score, 2)
-            break
+        if response_url.startswith("https://inspection.canada.ca"):
+            try:
+                response_number = int(re.findall(PATTERN, response_url)[0])
+                if response_number in expected_number:
+                    position = idx
+                    score = 1 - (position / total_pages)
+                    score= round(score, 2)
+                    break
+            except IndexError:
+                pass
```

Comment:
Why pass on IndexError? Is it because if re.findall doesn't find a list, the data is not want you need?

Reply:
Exactly, the `re.findall()` function does not always find a list, and as you correctly pointed out, if it's not a list, it means it's not a URL that I want to process.